### PR TITLE
[DOCS] Update enrollment token docs for multi-node configuration

### PR DESCRIPTION
**Pull request by lockewritesdocs**
Updates the documentation where we describe how to manually set up a multi-node cluster with an enrollment token. The updated section for enrolling nodes in an existing cluster is used in the installation documentation for both `tar.gz` and Windows, plus our documentation for starting the Stack with security enabled.

Relates to #80991

### CODE REVIEW

File: x-pack/docs/en/security/enroll-nodes.asciidoc [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+configuration.
+
+Before enrolling a new node, additional actions such as binding to an address
+other than `localhost` or satisfying bootstrap checks are typically necessary
+in production clusters. During that time, an auto-generated enrollment token
+could expire, which is why enrollment tokens aren't generated automatically.
+
+Additionally, only nodes on the same host can join the cluster without
+additional configuration. If you want nodes from another host to join your
+cluster, you need to set `transport.host` to `_site_`. Refer to
```
Comment (jkakavas) [inline x-pack/docs/en/security/enroll-nodes.asciidoc:16]:
```suggestion
cluster, you need to set `transport.host` to something other than `_local_` (i.e. `_site_`, or an IP address bound to an interface that is reachable by other hosts). Refer to 
```

Comment (lockewritesdocs) [reply x-pack/docs/en/security/enroll-nodes.asciidoc:16]: Thanks @jkakavas! I like your suggestion. Are you ok with the following change?

>If you want nodes from another host to join your cluster, you need to set `transport.host` to a supported value other than `_local_` (such as `_site_`), or an IP address that's bound to an interface where other hosts can reach it.

**Review by jkakavas — APPROVED** [review]

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Comment by lockewritesdocs** [comment]
@elasticmachine run ci/packaging-tests-windows-sample

**Comment by lockewritesdocs** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows-sample

**Pull request merged**